test/fit_tests/common/fit_common.py

```diff
@@ -703,18 +707,27 @@ def apply_obm_settings_new():
                             nodestatus[node]['status'] = "succeeded"
                         if state in ["failed", "cancelled", "timeout"]:
                             nodestatus[node]['status'] = "pending"
-        if VERBOSITY > 4:
+                            # if the workflow left an invalid OBM, delete it
+                            result = rackhdapi("/api/2.0/nodes/" + node)
+                            if result['status'] == 200:
+                                if result['json']['obms']:
+                                    for ref in result['json']['obms']:
+                                        obmref = ref.get('ref')
+                                        if obmref:
+                                            rackhdapi(obmref, action="delete")
+
+        if VERBOSITY >= 4:
             print "**** Node(s) OBM status:\n", json.dumps(nodestatus, sort_keys=True, indent=4,)
         if "pending" not in str(nodestatus) and "running" not in str(nodestatus):
             # All OBM settings successful
             return True
-        time.sleep(10)
+        time.sleep(30)
```

Comment:
This routine was designed to more efficiently set OBMs by running in parallel. The routine will complete when all of the workflows end successfully. If one or more node fails, they it will run for the full 15 minutes. The routine puts out a message into logs or to the console if running verbosity of 4 or greater that shows the node list, node ID, workflow ID and completion status on every retry.

Example:
**** Node(s) OBM status:
{
    "57ec10b586574e9007966338": {
        "instanceId": "c71ad907-99f2-4064-af93-bccb0f517241",
        "retry": 6,
        "sku": "Phoenix",
        "status": "succeeded"
    },
    "57ec10ee86574e900796633f": {
        "instanceId": "14b7f7fb-d3e0-4aed-9b42-dc9b73b287d6",
        "retry": 6,
        "sku": "Phoenix",
        "status": "running"
    },
    "57ec110586574e9007966346": {
        "instanceId": "b2dbe6b2-18bf-451c-bf9e-699c0006bcbe",
        "retry": 6,
        "sku": "Kylin",
        "status": "running"
    }
}

Here is the completion after 95 seconds:

**** Node(s) OBM status:
{
    "57ec10b586574e9007966338": {
        "instanceId": "c71ad907-99f2-4064-af93-bccb0f517241",
        "retry": 6,
        "sku": "Phoenix",
        "status": "succeeded"
    },
    "57ec10ee86574e900796633f": {
        "instanceId": "14b7f7fb-d3e0-4aed-9b42-dc9b73b287d6",
        "retry": 7,
        "sku": "Phoenix",
        "status": "succeeded"
    },
    "57ec110586574e9007966346": {
        "instanceId": "b2dbe6b2-18bf-451c-bf9e-699c0006bcbe",
        "retry": 7,
        "sku": "Kylin",
        "status": "succeeded"
    }
}
.
----------------------------------------------------------------------
Ran 1 test in 95.030s